tests/test_daily_temperature_extrema.py

```diff
@@ -0,0 +1,543 @@
+import datetime
+
+from pytest import fixture, mark, approx
+from sqlalchemy.sql import text
+
+from pycds.util import generic_sesh
+from pycds import Network, Station, History, Variable, Obs, NativeFlag, PCICFlag
+from pycds.weather_anomaly import DailyMaxTemperature, DailyMinTemperature
+
+
+def describe_function_effective__day__for__Tmax():
+    @mark.parametrize('tod, obs_time', [
+        ('morning', '2000-01-01 07:23'),
+        ('afternoon', '2000-01-01 16:18')
+    ])
+    @mark.parametrize('freq, expected_day', [
+        ('1-hourly', {'morning': datetime.datetime(2000, 1, 1), 'afternoon': datetime.datetime(2000, 1, 1)}),
+        ('12-hourly', {'morning': datetime.datetime(2000, 1, 1), 'afternoon': datetime.datetime(2000, 1, 2)})
+    ])
+    def it_returns_the_expected_day_of_observation(mod_empty_database_session, tod, obs_time, freq, expected_day):
```

Comment:
Can you add a comment about how tod == "time_of_day", just to save future devs some thinking time? :)